posts/tests.py

```diff
@@ -1,3 +1,312 @@
+
+
+@pytest.fixture
+def new_poll(new_user):
+    return Poll(author=new_user, description=DESCRIPTION)
+
+
+@pytest.fixture
+def new_pollFile(new_poll):
+    return PollFile(poll=new_poll, file=FILE1)
+
+
+@pytest.fixture
+def new_choice(new_poll):
+    return Choice(poll=new_poll, choice_text=CHOICETEXT1)
+
+
+# -------------------------------------------- Poll tests --------------------------------------------
+
+
+# Check that the Poll values are the same as the Poll inputs.
+@pytest.mark.django_db()
+def test_new_poll(new_poll):
+    assert new_poll.author.username == USERNAME
+    assert new_poll.author.first_name == FIRSTNAME
+    assert new_poll.author.last_name == LASTNAME
+    assert new_poll.author.password == PASSWORD
+    assert new_poll.author.email == EMAIL
+    assert new_poll.description == DESCRIPTION
+    assert new_poll.amount_of_voters == 0
+    assert len(new_poll.pollfile_set.all()) == 0
+    assert len(new_poll.choice_set.all()) == 0
+    new_poll.author.save()
+    assert new_poll.author.profile.profile_pic == 'profile-icon.png'
+
+
+# Check if the Poll is saved in the database and accessible via his author.
+@pytest.mark.django_db()
+def test_persist_poll(new_poll):
+    assert new_poll not in Poll.objects.all()
+    new_poll.author.save()
+    new_poll.save()
+    assert new_poll in Poll.objects.all()
+    assert new_poll in User.objects.filter(username=USERNAME).first().poll_set.all()
+
+
+# Check if Poll deletion delete only Poll from database.
+@pytest.mark.django_db()
+def test_delete_poll_aouthor(new_poll):
+    user = new_poll.author
+    user.save()
+    new_poll.save()
+    assert new_poll in Poll.objects.all()
+    assert user in User.objects.all()
+    new_poll.delete()
+    assert new_poll not in Poll.objects.all()
+    assert user in User.objects.all()
+
+
+# Check if Poll's author deletion delete both the author and and the poll.
+@pytest.mark.django_db()
+def test_delete_poll(new_poll):
+    user = new_poll.author
+    user.save()
+    new_poll.save()
+    assert new_poll in Poll.objects.all()
+    assert user in User.objects.all()
+    user.delete()
+    assert new_poll not in Poll.objects.all()
+    assert user not in User.objects.all()
+
+
+# verify that it is impossible to create a Poll with unvalid params.
+def test_poll_unvalid_args():
+    with pytest.raises(TypeError):
+        Poll(user=USERNAME, description=DESCRIPTION)
+
+
+# -------------------------------------------- PollFile tests --------------------------------------------
+
+
+# Check that the PollFile values are the same as the PollFile inputs.
+def test_new_pollfile(new_pollFile):
+    assert new_pollFile.poll.author.username == USERNAME
+    assert new_pollFile.poll.description == DESCRIPTION
+    assert new_pollFile.poll.amount_of_voters == 0
+    assert new_pollFile.file == FILE1
+
+
+# Check if the PollFile is saved in the database and accessible via his Poll.
```

Comment:
Fix the sentence in the note: instead of "..via his Poll.",
it's supposed to be: "...via it's Poll" or better "via the Poll it belongs to."

Reply:
My bad, its Poll and not it's Poll, the same applies on the similar review I wrote.